mkosi/tree.py

```diff
@@ -113,7 +121,10 @@ def copy_tree(
         attrs += ",timestamps,ownership"
 
         # Trying to copy selinux xattrs to overlayfs fails with "Operation not supported" in containers.
-        if statfs(os.fspath(dst.parent)) != OVERLAYFS_SUPER_MAGIC or not tree_has_selinux_xattr(src):
+        if statfs(os.fspath(dst.parent)) not in (
+            OVERLAYFS_SUPER_MAGIC,
+            ZFS_SUPER_MAGIC,
```

Comment:
I'm not sure this is the best solution. I think the problem comes from zfs just being such a foreign object with terrible non-Linux defaults. It should work with the zfs property `xattr=sa` set, which, if I remember correctly puts the xattr data into the inodes. That means, if this works on a properly configured zfs, that should be fixed on user side and put into our FAQ instead of handling zfs things explicity here.

Reply:
Oh ...

Mine is indeed not ```
xattr     on     inherited 
```

`xattr=on` is the default value though.

Reply:
Ok, so you seem to be running zfs 2.3, since `xattr=sa` is shown as `xattr=on`. 

Do you have any selinux labels in the first place? Probably, since you are running Fedora? In the issue you never mentioned what target distribution you were using. Does it support selinux? Are you using a tools tree? Honestly, zfs says it supports selinux (it even has some properties for it), if selinux-enabled tooling on a filesystem that purports to support selinux, apparently with the necessary options set to support selinux, is not able to set labels, that seems like a pretty outrageous bug on zfs side.

Reply:
I boot with `selinux=0` since first install and installed Fedora 41 through bootc.

It does not look like I have any selinux attributes to begin with.
```
❯  ls -laZ ~/.cache/.                    
total 508
drwxr-xr-x 32 sss users ?       40 Apr 17 12:45  .
drwx------ 38 sss users ?       56 Apr 17 12:32  ..
```